File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.js
function fac(n) {
    let result = 1;
    
    for (let i = 2; i <= n; i++) {
        result *= i;
    }
    
    return result;
}

function solution(balls, share) {
    return Math.round(fac(balls) / (fac(balls - share) * fac(share)));
} 

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.py
def solution(balls, share):
    balls_fac = 1
    balls_share_fac = 1
    share_fac = 1
    
    for i in range(balls, 0, -1):
        balls_fac *= i
        
    for j in range(balls-share, 0, -1):
        balls_share_fac *= j
        
    for k in range(share, 0, -1):
        share_fac *= k
        
    return balls_fac / (balls_share_fac * share_fac)